x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetector.java

```diff
@@ -86,6 +88,10 @@ public ExtractedFields detect() {
         if (extractedFields.getAllFields().isEmpty()) {
             throw ExceptionsHelper.badRequestException("No compatible fields could be detected in index [{}]", index);
         }
+        if (extractedFields.getAllFields().size() > docValueFieldsLimit) {
```

Comment:
Shouldn't this be `extractedFields.getDocValueFields().size()`?

Reply:
A few lines above we filter only doc_value fields. So it makes no difference.